### ISSUE TITLE
Extracting single band from multi-band areafile

### ISSUE THREAD

**Issue by ghansham** [visad/visad#138]
https://github.com/visad/visad/blob/19a77b3c852d13ceb9d9149d28e3df68057b10cf/core/src/visad/data/mcidas/AreaAdapter.java#L383

How to get a single band FlatField from a multiband areafile?
I think issues lies in this line, the second last arugment in the buildFlatField call should be band and not 0.
This will take care of both the cases when user wants a multiband extraction (band can be passed 0) and single band extraction (specific band number can be passed in this case).

**Comment by tommyjasmin** [comment]
I don't think so Ghansham.  Based on the comment, the intent of that block is to fetch all bands:

// getting the entire file including all bands

In this case the band parameter does need to be zero.

**Comment by ghansham** [comment]
That is fine but  there is no if else block which addresses the requirement
of extracting single band from multiband areafile.

**Comment by ghansham** [comment]
We should introduce an if else block to extract a single band. Because the
if else block in which band gets passed to buildFlatField function  also
wants numLines and numEles to be non-zero (see line no. 340)

Regards
Ghansham
